pkg/server/compression/compressor.go

```diff
@@ -0,0 +1,57 @@
+package compression
+
+import (
+	"compress/gzip"
+	"fmt"
+	"io"
+
+	"github.com/andybalholm/brotli"
+	"github.com/datadog/zstd"
+)
+
+type dummyCloser struct {
+	io.WriteCloser
+	w io.Writer
```

Comment:
```suggestion
	io.Writer
```
By embedding the writer, the Write function doesn't need to be re-implemented.

Reply:
I removed dummyCloser entirely, and added a small branch in the server to check the non-compression part. It's smaller and probably less confusing.